.github/workflows/build-providers.yaml

```diff
@@ -0,0 +1,57 @@
+name: Build release providers
+
+permissions:
+  id-token: write
+  contents: read
+  packages: write
+
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  providers-build:
+    runs-on: depot-ubuntu-22.04
+    concurrency:
+      group: oss-providers-build
+      cancel-in-progress: true
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Log in to GitHub Container Registry
+        uses: docker/login-action@v3
+        with:
+          registry: ghcr.io
+          username: ${{ secrets.GHCR_USERNAME }}
+          password: ${{ secrets.GHCR_TOKEN }}
```

Comment:
Do we need this? Can we use the token passed to the action?

Reply:
hmm.. we probably could get rid of these yeah. ill follow up on that in a separate set of prs